src/component/ProfileEditForm.tsx

```diff
@@ -23,11 +25,28 @@ interface ProfileEditFormProps {
 
 function ProfileEditForm({ onClose, onSave }: ProfileEditFormProps){
     const { register, formState: {errors}, handleSubmit } = useForm<ProfileEditFormInput>();
+    const updateUser = useUserStore(state => state.updateUser);
 
-    const onSubmit: SubmitHandler<ProfileEditFormInput> = (data) => {
+    const onSubmit: SubmitHandler<ProfileEditFormInput> = async (data, e) => {
+        e?.preventDefault();
+        const userData: User = {
+            id: 0,
+            name: data.username,
+            email: data.email,
+            experience: data.experience,
+            bio: data.bio,
+            gitHubLink: data.github,
+            skills: [], // You might not have skills in the form data
+        };
         console.log(data);
-        onSave(data);
-    }
+        console.log(userData);
+        await updateUser(userData).then(() => {
```

Comment:
There is no point in using both await and then. If you use "then" - remove await.